grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServiceImplErrorTest.java

```diff
@@ -0,0 +1,432 @@
+        channel.shutdownNow();
+        blockingChannel.shutdownNow();
+    }
+
+    @Test
+    void clientUnaryCall2() throws InterruptedException {
+        Throwable throwable = catchThrowable(() -> {
+            TestServiceBlockingStub service = TestServiceGrpc.newBlockingStub(blockingChannel);
+//                                                             .withDeadlineAfter(1, TimeUnit.SECONDS);
+            // Server impl throws error in the middle of transaction, client side should terminate connection
+            // as server cannot terminate connection.
+            service.unaryCall2(REQUEST_MESSAGE);
+        });
+        assertThat(Status.fromThrowable(throwable).getCode()).isEqualTo(Status.ABORTED.getCode());
+        assertThat(Status.trailersFromThrowable(throwable)).satisfies(metadata -> {
+            assertThat(metadata).isNotEqualTo(newCorruptedMutateMetaData(5, new byte[] { 1 }));
+        });
+        System.out.println(server.requestContextCaptor().size());
+        RequestLog log = server.requestContextCaptor().take().log().whenComplete()
+                               .join(); // todo: <- Never 'complete', so timeout. why?
+//         InternalMetadata#serialize causes ArrayIndexOutOfBoundsException.class
+        assertThat(log.responseCause()).isInstanceOf(StatusRuntimeException.class);
+        assertThat(log.responseCause()).isInstanceOf(ArrayIndexOutOfBoundsException.class);
+    }
+
+    @Test
+    void clientUnaryCall() throws InterruptedException {
+        Throwable throwable = catchThrowable(() -> {
+            TestServiceBlockingStub service = TestServiceGrpc.newBlockingStub(blockingChannel);
+//                                                             .withDeadlineAfter(1, TimeUnit.SECONDS);
+            // Server impl throws error in the middle of transaction, client side should terminate connection
+            // as server cannot terminate connection.
+            service.unaryCall(REQUEST_MESSAGE);
+        });
+        assertThat(Status.fromThrowable(throwable).getCode()).isEqualTo(Status.UNKNOWN.getCode());
+        assertThat(Status.trailersFromThrowable(throwable)).satisfies(metadata -> {
+            assertThat(metadata).isNotEqualTo(newCorruptedMutateMetaData(5, new byte[] { 1 }));
+        });
+
+        System.out.println(server.requestContextCaptor().size()); // todo: why 0 is returned?
+        RequestLog log = server.requestContextCaptor().take().log().whenComplete()
+                               .join(); //todo: <- Never 'complete', so timeout. why?
+//         InternalMetadata#serialize causes ArrayIndexOutOfBoundsException.class
+        assertThat(log.responseCause()).isInstanceOf(ArrayIndexOutOfBoundsException.class);
+    }
+
+    @Test
+    void clientUnaryCallServerStreamingOutputCall() throws InterruptedException {
+        Throwable throwable = catchThrowable(() -> {
+            TestServiceBlockingStub service = TestServiceGrpc.newBlockingStub(blockingChannel);
```

Comment:
I think once this is fixed, then streaming requests is not returned immediately and waits until timeout happens (by default 10 seconds). Can you also also check? 😄 

```suggestion
            TestServiceBlockingStub service = TestServiceGrpc.newBlockingStub(channel);
```

Reply:
Now all test ends like a charm.